tests/test_git_status.py

```
@@ -0,0 +1,144 @@
+                    if check:
+                        raise subprocess.CalledProcessError(1, cmd)
+                    return ""
+                elif "status --porcelain" in cmd:
+                    return "?? README.md"
+                return ""
+
+            mock_run.side_effect = side_effect
+
+            with patch("sys.stdout", new=StringIO()) as fake_out:
+                # Mock CalledProcessError since it's used in the function
+                with patch.object(subprocess, "CalledProcessError", Exception):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)